jetbrains-core/tst/software/aws/toolkits/jetbrains/services/schemas/search/SchemaSearchDialogTest.kt

```diff
@@ -100,188 +84,228 @@ class SchemaSearchDialogTest {
         }
 
-        future.thenApply {
-            assertSearchingDialogState(it)
+        searchingLatch.await(3, TimeUnit.SECONDS)
 
-            verify { mockSchemaSearchExecutor.searchSchemasAcrossAllRegistries(SEARCH_CRITERIA, any(), any()) }
+        try {
+            assertSearchingDialogState(dialog)
+        } finally {
+            waitForAsserts.countDown()
         }
     }
 
     @Test
     fun singleRegistrySearchWithResults() {
         val searchResults = listOf(SchemaSearchResult(SCHEMA1, VERSIONS))
 
-        every { mockSchemaSearchExecutor.searchSchemasInRegistry(REGISTRY1, SEARCH_CRITERIA, captureLambda(), any()) } answers
-            { lambda<OnSearchResultReturned<SchemaSearchResult>>().invoke(searchResults) }
-
-        val future = CompletableFuture<SchemaSearchSingleRegistryDialog>()
-        runInEdtAndWait {
-            val dialog = initSingleRegistryDialog()
+        val latch = CountDownLatch(1)
 
-            dialog.searchTextField.text = SEARCH_CRITERIA
-
-            future.complete(dialog)
+        mockSchemaSearchExecutor.stub {
+            on { searchSchemasInRegistry(eq(REGISTRY1), eq(SEARCH_CRITERIA), any(), any()) }.thenAnswer {
+                it.getArgument<OnSearchResultReturned<SchemaSearchResult>>(2).invoke(searchResults)
+                latch.countDown()
+            }
         }
 
-        future.thenApply {
-            assertSearchResultsDialogState(it, searchResults, emptyList())
+        val dialog = runInEdtAndGet {
+            initSingleRegistryDialog().also {
+                it.searchTextField.text = SEARCH_CRITERIA
+            }
         }
+
+        latch.await(3, TimeUnit.SECONDS)
+
+        assertSearchResultsDialogState(dialog, searchResults, emptyList())
     }
 
     @Test
     fun singleRegistrySearchError() {
         val searchError = SchemaSearchError(REGISTRY1, "someErrorMessage")
 
-        every { mockSchemaSearchExecutor.searchSchemasInRegistry(REGISTRY1, SEARCH_CRITERIA, any(), captureLambda()) } answers
-            { lambda<OnSearchResultError>().invoke(searchError) }
+        val latch = CountDownLatch(1)
 
-        val future = CompletableFuture<SchemaSearchSingleRegistryDialog>()
-        runInEdtAndWait {
-            val dialog = initSingleRegistryDialog()
-
-            dialog.searchTextField.text = SEARCH_CRITERIA
-
-            future.complete(dialog)
+        mockSchemaSearchExecutor.stub {
+            on { searchSchemasInRegistry(eq(REGISTRY1), eq(SEARCH_CRITERIA), any(), any()) }.thenAnswer {
+                it.getArgument<OnSearchResultError>(3).invoke(searchError)
+                latch.countDown()
+            }
         }
 
-        future.thenApply {
-            assertSearchResultsDialogState(it, emptyList(), listOf(searchError))
+        val dialog = runInEdtAndGet {
+            initSingleRegistryDialog().also {
+                it.searchTextField.text = SEARCH_CRITERIA
+            }
         }
+
+        latch.await(3, TimeUnit.SECONDS)
```

Comment:
nit: since this is copy pasted, why not have a wait function if we need to change the timeout ever